climada/hazard/centroids/centr.py

```diff
@@ -79,280 +52,215 @@ class Centroids():
-        """Create Centroids object with meta attribute according to pixel border data.
-
-        .. deprecated:: 3.3
-           This method will be removed in a future version. CLIMADA will only support
-           regular grids with a constant lat/lon resolution then. Use
-           :py:meth:`from_pnt_bounds` instead.
+    def from_exposures(cls, exposures):
+        """Generate centroids from the location of an exposures.
 
         Parameters
         ----------
-        xf_lat : float
-            upper latitude (top)
-        xo_lon : float
-            left longitude
-        d_lat : float
-            latitude step (negative)
-        d_lon : float
-            longitude step (positive)
-        n_lat : int
-            number of latitude points
-        n_lon : int
-            number of longitude points
-        crs : dict() or rasterio.crs.CRS, optional
-            CRS. Default: DEF_CRS
+        exposures : Exposure
+            exposures from which to take the centroids location
+            and region_id (if defined) and on_land (if defined)
 
         Returns
         -------
-        centr : Centroids
-            Centroids with meta according to given pixel border data.
-        """
-        LOGGER.warning(
-            "Centroids.from_pix_bounds has been deprecated and will be removed in a "
-            "future version. Use Centroids.from_pnt_bounds instead."
-        )
-
-        meta = {
-            'dtype': 'float32',
-            'width': n_lon,
-            'height': n_lat,
-            'crs': crs,
-            'transform': rasterio.Affine(d_lon, 0.0, xo_lon, 0.0, d_lat, xf_lat),
-        }
-
-        return cls(meta=meta)
-
-    def set_raster_from_pnt_bounds(self, *args, **kwargs):
-        """This function is deprecated, use Centroids.from_pnt_bounds instead."""
-        LOGGER.warning("The use of Centroids.set_raster_from_pnt_bounds is deprecated. "
-                       "Use Centroids.from_pnt_bounds instead.")
-        self.__dict__ = Centroids.from_pnt_bounds(*args, **kwargs).__dict__
+        Centroids
+            Centroids built from the exposures
+        """
+        col_names = [
+            column
+            for column in exposures.gdf.columns
+            if column in ['region_id', 'on_land']
+            ]
+
+        # Legacy behaviour
+        # Exposures can be without geometry column
+        #TODO: remove once exposures is real geodataframe with geometry.
+        if 'geometry' in exposures.gdf.columns:
+            col_names += ['geometry']
+            gdf = exposures.gdf[col_names]
+            return cls.from_geodataframe(gdf)
+
+        if 'latitude' in exposures.gdf.columns and 'longitude' in exposures.gdf.columns:
+            gdf = exposures.gdf[col_names]
+            return cls(
+                lat = exposures.gdf['latitude'],
+                lon = exposures.gdf['longitude'],
+                crs = exposures.crs,
+                **dict(gdf.items())
+            )
```

Comment:
No fallback `return` (this is the source of the `inconsistent-return-statements` linter issue)

Reply:
Done in [1f6ec54](https://github.com/CLIMADA-project/climada_python/pull/787/commits/1f6ec54f6aa3891b67abc105d44c57ffd526b064)